packages/experimental/Badge/SPEC.md

```diff
@@ -0,0 +1,330 @@
+Basic Badges do not recieve focus and are not accessible. Badges should be given a meaningful description.
+This description will be applied, via `accessibilityLabel` to the element decorated by the Badge.
+
+## PresenceBadge
+
+Presence badge represents user's availability.
+
+### PresenceBadge Props
+
+```ts
+export interface PresenceBadgeProps extends BadgeCoreProps {
+  status?: PresenceBadgeStatus;
+  outOfOffice?: boolean;
+}
+```
+
+### PresenceBadge Tokens
+
+```ts
+export interface PresenceBadgeTokens extends BadgeCoreTokens {
+  available?: PresenceBadgeTokens;
+  away?: PresenceBadgeTokens;
+  awayOutOfOffice?: PresenceBadgeTokens;
+  offline?: PresenceBadgeTokens;
+  outOfOffice?: PresenceBadgeTokens;
+  doNotDisturb?: PresenceBadgeTokens;
+  busy?: PresenceBadgeTokens;
+  unknown?: PresenceBadgeTokens;
+  blocked?: PresenceBadgeTokens;
+  status?: PresenceBadgeStatus;
+}
+```
+
+## CounterBadge
+
+Small numerical value or status descriptor for UI elements.
+
+### CounterBadge States
+
+The following section describes the different states which `CounterBadge` can be in as a result of interaction.
+
+#### showBadge state
+
+`showBadge` is set to true when `count` is bigger than 0, or when showZero is set to `true`.
+
+#### displayCount state
+
+`displayCount` state has content to show,e.g. if `count` is 1500 and `overflowCount` is 1000, then `displayCount` is `1000+`.
+
+### Interaction
+
+1. By default badge will be hidden when count is 0.
+2. If count is greater than zero badge is shown.
+3. Use property `showZero` to show badge even if count is at zero.
+
+#### Overflow count
+
+`[overflow count]+` is displayed when count is larger than the limit of displayed digits. Limit is set by prop `overflowCount`.
+Examples:
+By default `overflowCount` is equal to 99. If `count` is larger then `99+` will be shown.
+
+### CounterBadge props
+
+```ts
+export interface CounterBadgeProps extends Omit<BadgeCoreProps, 'appearance' | 'color'>, Omit<BadgeProps, 'shape'> {
+  /**
+   * A Badge can have different appearances that emphasize certain parts of it:
+   *  - filled: The default appearance if one is not specified.
+   *    The badge background is filled with color with a contrasting foreground text to match.
+   *  - ghost: The badge background is transparent, with the foreground text taking color to emphasize it.
+   * @default filled
+   */
+  appearance?: 'filled' | 'ghost';
+
+  /**
+   * Semantic colors for a counter badge
+   * @default brand
+   */
+  color?: Extract<BadgeProps['color'], 'brand' | 'danger' | 'important' | 'informative'>;
+
+  /**
+   * Value displayed by the Badge
+   * @default 0
+   */
+  count?: number;
+
+  /**
+   * If a dot should be displayed without the count
+   * @default false
+   */
+  dot?: boolean;
+
+  /**
+   * Max number to be displayed
+   * @default 99
+   */
+  overflowCount?: number;
+
+  /**
+   * A Badge can be circular or rounded
+   * @default circular
+   */
+  shape?: 'circular' | 'rounded';
```

Comment:
Does this badge not support square?

Reply:
Basic Badge support square, but no CounterBadge